py3status/modules/arch_updates.py

```diff
@@ -135,6 +138,20 @@ def _check_aur_updates(self):
             return str(pending_updates).count(LINE_SEPARATOR)
         return None
 
+    def _check_aur_updates_yay(self):
+        """
+        This method will use the 'yay' command line utility
+        to determine how many updates are waiting to be installed
+        from the AUR.
+        Returns: None if unable to determine number of pending updates
+        """
+        try:
+            pending_updates = str(subprocess.check_output(
+                ["yay", "--query", "--upgrades", "--aur"]))
```

Comment:
(Question) `$ man yay` have this...
```
       -u, --upgrades
              Deprecated, use yay -Qu instead.
```
I'm not sure what `-u` is referring to... I'm guessing pacman's `-u, --upgrades`. It looks like we don't have to change anything, but to change from `--upgrades` to `-u` just in case `yay` removed `--upgrades` someday? I don't know what's being deprecated exactly... because pacman have `--upgrades` too.

I wonder if there is a `yay` command to get a number of pacman packages instead of using `checkupdates`.

Reply:
Your quote comes from the "YAY OPTIONS (APPLY TO -Y AND --YAY)" section and thus not applies to the `--query` main command being used. `yay --query --upgrades --aur` is the long form for `yay -Qu --aur`. Which do you prefer?

Concerning Arch updates, yes and no: `yay --query --upgrades --arch` ouputs the update-able Arch packages, but requires `sudo pacman -Sy` to be run first to give meaningful results.

Reply:
Ah, sorry, I am trying to get familiar with `yay`. I see it now in `SHOW OPTIONS (APPLY TO -P AND --SHOW)`.  I prefer the (current) long argument.

Thanks for letting me know about `sudo pacman -Sy` for `yay`. I put out https://github.com/ultrabug/py3status/pull/1552 and wanted to see if we could use `yay` (again) instead of `checkupdates`.

Reply:
I think using `checkupdates` would be preferable for packages in official repos, don't completely lose the support for it in #1552 😉

Reply:
Not removing or replacing. I meant to say: Make `{yay_arch}` and `{yay_aur}` possible so if we have `yay` installed (but not `checkupdates`), then it could be like that. If we have both installed, then it would be all three on the bar due to checked binaries.

I'm looking at aur helpers and will be adding `auracle` support next. :astonished: My expectation is that users don't go and install bunch of aur helpers, but their favorite ones... and the bar would be only displaying (optional) checkupdates and/or (optional) favorite aur helper.